setup.py

```diff
@@ -14,7 +14,7 @@
         url='https://github.com/di/pytest-reqs',
         py_modules=['pytest_reqs'],
         entry_points={'pytest11': ['reqs = pytest_reqs']},
-        install_requires=['pytest>=2.4.2', 'packaging>=17.1'],
+        install_requires=['pip<10', 'pytest>=2.4.2', 'packaging>=17.1'],
```

Comment:
This will cause any version of `pip>=10` to be uninstalled and `pip<10` to be installed when installing `pytest-reqs`.

I'm not sure I want this behavior -- I think it's a bad practice to surreptitiously downgrade other libraries, and I think the decision to do so should be left to the user rather than doing it for them.

One thing we could do is move the warning into `setup.py` instead and fail the installation if `pip>=10`, so the user gets an error a little earlier, and doesn't end up installing `pytest-reqs` if it's going to be incompatible.

That said, I plan to support `pip>=10` in the near future, so I think this is probably fine-as for the time being until that support is added, at which point the warning will no longer be necessary.

Reply:
All packages should declare their dependencies.  It is basic metadata.
pip doesnt do a good job of alerting users to conflicting sets of requirements, but .. hey .. that is what pytest-reqs fixes in test environments!

Needless to say, if anyone is somehow installing `pytest-reqs` into a production environment, they deserve for it to break. 

If you plan to add pip 10 support before the next release, then this is just a CI fix, and it is the simplest one.

fwiw, if you do `pip>=10` in the next release, that will break for @coala which uses 'pip<10' in our CI because of other packages (e.g. `safety`) which are also still broken by pip 10, but we dont include that pin in our release so our users can use whatever `pip` or `safety` etc that they need to use.
Ideally this pytest plugin supports old and new pip, at least for a while until all popular packages-which-poked-inside-pip have been fixed.